data-access-layer/eventmesh/ApiServerClient.js

```diff
@@ -59,6 +73,61 @@ class ApiServerClient {
     });
   }
 
+  /**
+   * Poll for Status until opts.start_state changes
+   * @param {object} opts - Object containing options
+   * @param {string} opts.operationId - Id of the operation ex. backupGuid
+   * @param {string} opts.start_state - start state of the operation ex. in_queue
+   * @param {object} opts.started_at - Date object specifying operation start time
+   */
+  getResourceOperationStatus(opts) {
+    logger.info(`Waiting ${CONST.EVENTMESH_POLLER_DELAY} ms to get the operation state`);
+    let finalState;
+    return Promise.delay(CONST.EVENTMESH_POLLER_DELAY)
+      .then(() => this.getOperationState({
+        operationName: CONST.OPERATION_TYPE.BACKUP,
+        operationType: CONST.APISERVER.RESOURCE_TYPES.DEFAULT_BACKUP,
+        operationId: opts.operationId
+      })).then(state => {
+        const duration = (new Date() - opts.started_at) / 1000;
+        logger.info(`Polling for ${opts.start_state} duration: ${duration} `);
+        if (duration > CONST.BACKUP.BACKUP_START_TIMEOUT) {
+          logger.error(`Backup with guid ${opts.operationId} not picked up from the queue`);
+          throw new Timeout(`Backup with guid ${opts.operationId} not picked up from the queue`);
```

Comment:
should you also not update the status of the backup as failed and ensure that backup never gets picked up by manager in background?

Reply:
Also, shouldnt this check move to the else clause in LN112? in the last iteration assuming on boundary condition u r really timing out and it was processed (either in-progress or error), but you would be instead responding back as Timedout?

Reply:
As you have suggested the backup start is now asynchronous.

Reply:
the check for timeout is made after the state check in the upcoming PR https://github.com/cloudfoundry-incubator/service-fabrik-broker/compare/master...muralisc:add-error-field?expand=1#diff-db2af8e82ceb9a902cfd3daa0b0706cfR112